source/patterns/@aws-solutions-constructs/aws-iot-sqs/README.md

```diff
@@ -0,0 +1,101 @@
+> This means that while you may use them, you may need to update your source code when upgrading to a newer version of this package.
+
+---
+<!--END STABILITY BANNER-->
+
+| **Reference Documentation**:| <span style="font-weight: normal">https://docs.aws.amazon.com/solutions/latest/constructs/</span>|
+|:-------------|:-------------|
+<div style="height:8px"></div>
+
+| **Language**     | **Package**        |
+|:-------------|-----------------|
+|![Python Logo](https://docs.aws.amazon.com/cdk/api/latest/img/python32.png) Python|`aws_solutions_constructs.aws_iot_sqs`|
+|![Typescript Logo](https://docs.aws.amazon.com/cdk/api/latest/img/typescript32.png) Typescript|`@aws-solutions-constructs/aws-iot-sqs`|
+|![Java Logo](https://docs.aws.amazon.com/cdk/api/latest/img/java32.png) Java|`software.amazon.awsconstructs.services.iotsqs`|
+
+This AWS Solutions Construct implements an AWS IoT MQTT topic rule and an AWS SQS Queue pattern.
+
+Here is a minimal deployable pattern definition in Typescript:
+
+``` typescript
+const { IotToSqsProps, IotToSqs } from '@aws-solutions-constructs/aws-iot-sqs';
+
+const props: IotToSqsProps = {
+    iotTopicRuleProps: {
+        topicRulePayload: {
+            ruleDisabled: false,
+            description: "Testing the IotToSqs Pattern",
+            sql: "SELECT * FROM 'iot/sqs/#'",
+            actions: []
+        }
+    }
+};
+
+new IotToSqs(this, 'test-iot-sqs-integration', props);
+```
+
+## Initializer
+
+``` text
+new IotToSqs(scope: Construct, id: string, props: IotToSqsProps);
+```
+
+_Parameters_
+
+* scope [`Construct`](https://docs.aws.amazon.com/cdk/api/latest/docs/@aws-cdk_core.Construct.html)
+* id `string`
+* props [`IotToSqsProps`](#pattern-construct-props)
+
+## Pattern Construct Props
+
+| **Name**     | **Type**        | **Description** |
+|:-------------|:----------------|-----------------|
+|existingQueueObj?|[`sqs.Queue`](https://docs.aws.amazon.com/cdk/api/latest/docs/@aws-cdk_aws-sqs.Queue.html)|Existing instance of SQS queue object, providing both this and `queueProps` will cause an error.|
+|queueProps?|[`sqs.QueueProps`](https://docs.aws.amazon.com/cdk/api/latest/docs/@aws-cdk_aws-sqs.QueueProps.html)|User provided props to override the default props for the SQS queue.|
+|deadLetterQueueProps?|[`sqs.QueueProps`](https://docs.aws.amazon.com/cdk/api/latest/docs/@aws-cdk_aws-sqs.QueueProps.html)|Optional user provided properties for the dead letter queue.|
+|deployDeadLetterQueue?|`boolean`|Whether to deploy a secondary queue to be used as a dead letter queue. Default `true`.|
+|maxReceiveCount?|`number`|The number of times a message can be unsuccessfully dequeued before being moved to the dead-letter queue. Required field if `deployDeadLetterQueue`=`true`.|
+|enableEncryptionWithCustomerManagedKey?|`boolean`|Use a KMS Key, either managed by this CDK app, or imported. If importing an encryption key, it must be specified in the `encryptionKey` property for this construct.|
```

Comment:
This varies from the design guidelines for SQS and it's not clear what it accomplishes. encryptionKey already allows a client to specify an existing key to encrypt the queue.

Reply:
When creating the queue, I was planning to use the `buildQueue` SQS helper provided by Solutions Constructs. In [this snippet](https://github.com/awslabs/aws-solutions-constructs/blob/33a9dd7408d163ed07305b5bfaa2dcdd295207ac/source/patterns/%40aws-solutions-constructs/core/lib/sqs-helper.ts#L82-L90) it seems like the logic is that if `enableEncryptionWithCustomerManagedKey` is `false`, the queue will not have encryption enabled. If `true` and the user provides a key in `encryptionKey`, that one will be used. Otherwise a new key will be built.

Reply:
I was also using the `aws-sns-sqs` pattern as a guideline, which does [something similar](https://github.com/awslabs/aws-solutions-constructs/blob/33a9dd7408d163ed07305b5bfaa2dcdd295207ac/source/patterns/%40aws-solutions-constructs/aws-sns-sqs/lib/index.ts#L123-L130).

Reply:
Upon further review, I'm inclined to agree with you. I believe it's the design guidelines that need to be updated.